### PR TITLE
We Like New Functions

### PR DESCRIPTION
**Pull request by latot**
Hi again, well this time i have this functions, about the zero functions and related, i try avoid the most posible execute this line ```y = sym(eye(cell2nosyms(varargin){:}));```, because this is fast ```eye(cell2nosyms(varargin){:})```, but transform that to sym takes to much time with big matrix.

well other problem is, although i add the cell2sym function when i try execute it i get an error:
```
>> cell2sym({sym(1)})
error: 'cell2sym' undefined near line 1 column 1
```
and actually divisors depends of it, if still don't works maybe at least add it to private.

This Fixes https://github.com/cbm755/octsympy/issues/405 and add some functions of https://github.com/cbm755/octsympy/issues/215
Well. Cya.

### CODE REVIEW

**Comment by latot** [comment]
from travis:
```
>>>>> processing @sym/cell2sym.m
***** test
 A = {1, 2, 3;4, 5, 6;7, 8, 9};
 B = [1, 2, 3;4, 5, 6;7, 8, 9];
 assert( isequal( cell2sym(A), sym(B)))
!!!!! test failed
'cell2sym' undefined near line 5 column 19
```
D:

**Comment by latot** [comment]
moved the problem of cell2sym to https://github.com/cbm755/octsympy/pull/527

**Comment by latot** [comment]
mm, the same problem as cell2sym:
```
>>>>> processing @sym/diff2.m
***** test
 % octave's vector difference still works
 assert(isempty(diff2(1)))
 assert((diff2([2 6]) == 4))
!!!!! test failed
'diff2' undefined near line 4 column 17
shared variables 
  scalar structure containing the fields:
```

File: inst/@sym/charpoly.m [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod @@sym charpoly (@var{x})
+%% @defmethod @@sym charpoly (@var{x}, @var{y})
```
Comment (cbm755) [inline inst/@sym/charpoly.m]: maybe `A` and `x` (or `var`) are better names here?  Can you also file an issue for the missing vector output?

File: inst/@sym/charpoly.m [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+%% Characteristic polynomial of matrix.
+%%
+%% Example:
+%% @example
+%% @group
+%% syms x
+%% a = x*x;
```
Comment (cbm755) [inline inst/@sym/charpoly.m]: remove `a = `?

File: inst/@sym/charpoly.m [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+%!test
+%! syms x
+%! A = sym([x, x;x, x]);
+%! assert( isequal( charpoly(A, x), -x^2))
+
+%!xtest
+%! syms x
+%! A = sym([1, 2;3, 4]);
+%! assert( isequal( charpoly(A, x), x^2 - 5*x -2))
```
Comment (cbm755) [inline inst/@sym/charpoly.m]: Maybe this is a good doctest as well?   The doctest you have is a bit confusing for newbie.  How about using `sym('lambda')` or `mu` as the variable?  At least in my linalg classes, it was usually one of those...

File: inst/@sym/dawson.m [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+%% @seealso{@@sym/erfc, @@sym/erf, @@sym/erfcx, @@sym/erfi, @@sym/erfinv, @@sym/erfcinv}
+%% @end defmethod
+
+
+function y = dawson(x)
+  if (nargin ~= 1)
+    print_usage ();
+  end
+  y = uniop_helper (x, 'lambda a: exp(-a**2)*erfi(a)*(sqrt(S(pi))/2)');
+end
```
Comment (cbm755) [inline inst/@sym/dawson.m]: this needs a test, maybe something like:
````
octave:17> A = dawson([1 2])
A =

   0.53808   0.30134

octave:18> B = double(dawson(sym([1 2])))
B =

   0.53808   0.30134

octave:19> assert(A, B, -eps)
````

File: inst/@sym/hilb.m [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod @@sym hilb (@var{x})
+%% Return the Hilbert matrix of order N.
```
Comment (cbm755) [inline inst/@sym/hilb.m]: x versus N: not consistent

File: inst/@sym/invhilb.m [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod @@sym invhilb (@var{x})
+%% Return the inverse of the Hilbert matrix of order N.
```
Comment (cbm755) [inline inst/@sym/invhilb.m]: as above

File: inst/@sym/angle.m [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod @@sym angle (@var{x})
+%% Symbolic polar angle.
+%%
+%% Example:
+%% @example
+%% @group
+%% x = sym(2+3*i);
+%% y =  angle(x)
```
Comment (cbm755) [inline inst/@sym/angle.m]: remove extra space

File: inst/@sym/beta.m [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  if (nargin ~= 2)
+    print_usage ();
+  end
+
+  r = binop_helper(sym(x), sym(y), 'beta');
+
+end
+
+%!test
+%! assert( isequal( vpa(beta(1, 2)), sym(1)/2))
```
Comment (cbm755) [inline inst/@sym/beta.m]: does this pass?
`assert( isequal( double(beta(1, 2)), 1/2)`

File: inst/@sym/eye.m [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <http://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @documentencoding UTF-8
+%% @defmethod @@sym eye (@var{x})
+%% @defmethod @@sym eye (@var{x}, @var{y})
```
Comment (cbm755) [inline inst/@sym/eye.m]: maybe these could be `@var(n), @var(m)`?

**Comment by cbm755** [comment]
This looks great!  Some minor comments above.

I don't really understand your `eye`, `ones` and `zeros` yet.  Are you trying to support calls like:
````
zeros(sym(3), 2, 'double')
````
If so, those should be tested (and I don't think it works right now).  Or is it for something else?

If easy, you could split off those functions and do a new PR and we can discuss.  Then I can merge the other bits!

File: inst/@sym/beta.m (hunk 8 above)
Comment (latot) [inline inst/@sym/beta.m]: yes, it pass.

**Comment by latot** [comment]
okidokis, all proposed changes applied! zeros and similar to this pr https://github.com/cbm755/octsympy/pull/532.

**Pull request merged**